src/uwtools/cli.py

```diff
@@ -679,6 +745,24 @@ def _add_arg_leadtime(group: Group, required: bool = False) -> None:
     )
 
 
+def _add_arg_module(group: Group) -> None:
+    group.add_argument(
+        _switch(STR.module),
+        help="Name of driver module",
```

Comment:
How would the team feel about adding a bit extra help here to read "Name of driver module (the filename that holds the class, but without the py suffix)" or something like it?

I feel like especially in NWP the word module is overloaded and I've answered this question a lot of times when discussing with developers.

Interestingly, if I provide the full path here (relative and including the py suffix), it still works. Is that by design or accidental? Do we need the `--module-dir` arg?

```
...uwtools/src/uwtools/tests $ uw invoke --classname TestDriver --module ./fixtures/testdriver.py --task eighty_eight --schema-file ./fixtures/testdriver.jsonschema --cycle 2023-05-02 -c ./fixtures/simple.yaml --leadtime 20
[2024-07-22T08:26:56]     INFO 0 UW schema-validation errors found
[2024-07-22T08:26:56]     INFO 88: Initial state: Ready
```

Reply:
Interesting. It might be a nice simplification to eliminate `--module-dir` and have just `--module` (and its API equivalent) and have the value be a module name, with or without the `.py` extension. As a user, I'd hope for my module load be loaded

1. If I give `/path/to/foo.py`, `../other/dir/foo.py`, `./foo.py`, etc. -- any explicit path should work.
2. If I give `foo.py` and `foo.py` exists in my current directory.
3. If I give `foo` and the path containing `foo.py` exists on `PYTHONPATH`.

I think our current code implements 1 and 2, except that it would not need to append `.py` because the user would supply it. For 3, I think we might just need a fallback to try `importlib.import_module()`, which consults `sys.path` / `PYTHONPATH` automatically.

@elcarpenterNOAA What do you think about this? If it also seems like good and simpler semantics to you, and if you want to talk about the implementation, please let me know.

Reply:
If we implemented that, the help message might read
> .py filename or path, or name of module on sys.path

I think most users will use the `.py` form, but IMO the other option, which honors the default `sys.path` as well as `PYTHONPATH`, should exist for some power-user use cases.